src/Util/CharacterSort.ts

```diff
@@ -4,42 +4,45 @@ import i18n from "../i18n";
 import { initCharMeta } from "../stateInit";
 import { CharacterKey } from "../Types/consts";
 import { FilterConfigs, SortConfigs } from "./SortByFilters";
-export const characterSortKeys = ["new", "level", "rarity", "name",] as const
+export const characterSortKeys = ["new", "level", "rarity", "name", "favorite"] as const
 export type CharacterSortKey = typeof characterSortKeys[number]
 
 export function characterSortConfigs(database: ArtCharDatabase, characterSheets: Record<CharacterKey, CharacterSheet>): SortConfigs<CharacterSortKey, CharacterKey> {
   return {
     new: {
       getValue: (ck) => database.chars.get(ck as CharacterKey) ? 0 : 1,
-      tieBreaker: "name"
+      sortByKeys: ["new", "favorite", "name"]
     },
     name: {
       getValue: (ck) => i18n.t(`charNames_gen"${ck}`).toString(),
+      sortByKeys: ["name"]
     },
     level: {
       getValue: (ck) => {
         const char = database.chars.get(ck as CharacterKey)
         if (!char) return 0
         return char.level * char.ascension
       },
-      tieBreaker: "rarity"
+      sortByKeys: ["level", "rarity", "name"]
     },
     rarity: {
       getValue: (ck) => characterSheets?.[ck]?.rarity,
-      tieBreaker: "level"
+      sortByKeys: ["rarity", "level", "name"]
```

Comment:
for `name`, `level`, `rarity`, you can do `["favorite",...]` to make them sort by favorite 1st

Reply:
I wonder if it makes sense for these sorts to be applied from the UI side, instead of being hard-coded from the API side...